src/pivot-table/hooks/use-column-width.ts

```diff
@@ -1,167 +1,245 @@
 import { memoize } from "qlik-chart-modules";
 import { useCallback, useMemo } from "react";
 import { PSEUDO_DIMENSION_INDEX } from "../../constants";
-import type { LayoutService, LeftDimensionData, Rect, VisibleDimensionInfo } from "../../types/types";
+import {
+  ColumnWidthType,
+  type ColumnWidth,
+  type ExtendedDimensionInfo,
+  type ExtendedMeasureInfo,
+} from "../../types/QIX";
+import type { LayoutService, Rect, VisibleDimensionInfo } from "../../types/types";
 import { GRID_BORDER } from "../constants";
 import { useStyleContext } from "../contexts/StyleProvider";
 import useMeasureText from "./use-measure-text";
 
 interface ColumnWidthHook {
   leftGridWidth: number;
   rightGridWidth: number;
-  totalMeasureInfoColumnWidth: number;
-  getLeftColumnWidth: (index: number) => number;
-  getDataColumnWidth: (index: number) => number;
-  getMeasureInfoWidth: (index: number) => number;
-  getTotalWidth: () => number;
+  totalWidth: number;
+  showLastRightBorder: boolean;
+  getLeftGridColumnWidth: (index: number) => number;
+  getRightGridColumnWidth: (index?: number) => number;
 }
 
 export const EXPAND_ICON_WIDTH = 30;
-const MIN_COLUMN_WIDTH = 100;
-const MAX_RATIO_OF_TOTAL_WIDTH = 0.75;
+const LEFT_GRID_MAX_WIDTH_RATIO = 0.75;
+
+export enum ColumnWidthValues {
+  PixelsMin = 30,
+  PixelsMax = 7680,
```

Comment:
why we should allow this much size on a col? is there any reason why that number?

Reply:
It's the width of an 8k display. Used the same as for sn-table. My reasoning was that if you for some reason wanted a single column table, that takes up the entire screen, this is the width you need. It is pretty arbitrary, could lower it